tiny_dnn/core/framework/tensor.h

```diff
@@ -335,7 +318,57 @@ class Tensor {
                                                           shape_.end()));
     }
 
+    Tensor view(std::initializer_list<size_t> const &new_shape) {
+        return view_impl({}, new_shape);
+    }
+
+    Tensor view(std::initializer_list<size_t> const &start,
+                std::initializer_list<size_t> const &new_shape) {
+	return view_impl(start, new_shape);
+    }
+
+    /*
+     * @brief Returns whether the tensor is a view of another tensor
+     *
+     */
+    bool isView() const {
+        return size_ != storage_ptr_->size();
+    }
+
 private:
+    /**
+     * Constructor that accepts a pointer to existing TensorStorage, together
+     * with shape and offset.
+     * @param storage pointer to TensorStorage
+     * @param offset offset from first element of storage
+     * @param shape shape of the Tensor
+     * @return
+     */
+    explicit Tensor(const TensorStoragePointer storage,
+                    const size_t offset,
+		    std::initializer_list<size_t> const &shape) {
+        offset_ = offset;
+        size_ = product(shape);
+        storage_ptr_ = storage;
+        std::copy(shape.begin(), shape.end(), shape_.begin());
+    }
+
+    Tensor view_impl(std::initializer_list<size_t> const &start,
+                     std::initializer_list<size_t> const &new_shape) {
+	if (start.size() > kDimensions || new_shape.size() > kDimensions) {
+            throw nn_error("Overpassed number of existing dimensions.");
+	}
+
+	// compute the new offset and check that it's feasible to create
+	// the new view.
+	const size_t new_offset = compute_offset(start, shape_);
```

Comment:
Ah, ok, I misunderstood the meaning of `start` and the way `compute_offset` works. Please add some explanation to the function.